WcaOnRails/app/models/competition.rb

```diff
@@ -640,7 +640,11 @@ def in_progress?
   end
 
   def is_over?
-    !end_date.nil? && end_date < Date.today
+    # We treat "today == last day of competition" as the competition is over.
+    # Once we know the schedules and timezones of competitions, we'll be able
+    # to do something more accurate here.
+    # See https://github.com/thewca/worldcubeassociation.org/issues/737
+    !end_date.nil? && end_date <= Date.today
```

Comment:
> Isn't the model is now incorrect, since is_over? returns true before the competition is actually over?

Well, that's just the thing. We just don't know when competitions end, so we really have 3 choices:

1. Change the method to aggressively declare competitions over so that even if a competition in [UTC+14](https://en.wikipedia.org/wiki/UTC%2B14:00) (the furthest ahead time zone) is over, we declare it over. My change in this PR makes us pretty aggressive. However, if a competition at UTC+14 ends at 1pm (1300) on Sunday, then it will only be 11pm (2300) in UTC (our server) time and we will not yet declare the competition over, even though it is over.
2. Change the method to be very paranoid about declaring competitions over (basically, if a competition is scheduled to end on Sunday, wait until it is Monday EVERYWHERE before declaring the competition over). I was tickled to discover there's a term for this used in balloting procedures: [Anywhere on Earth (AoE)](https://en.wikipedia.org/wiki/Anywhere_on_Earth).
3. Do something in between. That's what we do right now. For example, the current code in `master` is comparing to UTC midnight, which sometimes declares a competition over before it is over (e.g. a competition that is still ongoing at 6pm in California on Sunday will actually be treated as ongoing, because it is already 6pm+8hrs = 2am UTC on Monday).

So, this is really a matter of tradeoffs, and I think requires us to look at all our uses of `is_over?` to make a decision. Here's a quick analysis of them:

- `app/assets/javascripts/application.js:91:    if (c.is_over) {` - This just determines what color competition markers show up as on maps when we show maps of them. IMO, not worth stressing over.
- `app/controllers/competitions_controller.rb:380:    @past_competitions, @not_past_competitions = competitions.partition(&:is_over?)` - This controls the behavior of the "my competitions" page (we show past competitions below the fold). I don't believe there's any good solution here.
- `app/models/competition.rb:169:  validate :delegates_must_be_delegates, unless: :is_over?` - Pretty ignorable, IMO. This is only relevant if the delegate of a competition is demoted right around the same time as their competition ends, and someone tries to edit the competition. Those are both vanishly rare events that are not likely to coincide, and even if they do, the behavior isn't too weird any way we choose to define `is_over`.
- `app/models/competition.rb:199:    if !self.results_posted? && self.is_over?` - This one is a bit more interesting. When you view a competition page, we present you with 1 of 2 interesting messages: either ["This competition is over, we are working to upload the results as soon as possible!"](https://github.com/thewca/worldcubeassociation.org/blob/097343c37844d7318613baef1913bdee53c5bce1/WcaOnRails/config/locales/en.yml#L642), or ["This competition is ongoing. Come back after %{date} to see the results!"](https://github.com/thewca/worldcubeassociation.org/blob/097343c37844d7318613baef1913bdee53c5bce1/WcaOnRails/config/locales/en.yml#L643). I don't feel too strongly about which message we show in which cases.
- `app/models/competition.rb:642:  def is_over?` - This is the controversial method we're talking about =)
- `app/models/delegate_report.rb:45:    competition.is_over?` - This is the definition of the `can_be_posted?` method for `DelegateReport`. Very much part of the controversy we're discussing right now =) As @SAuroux has requested, it's better to err on the side of prematurely declaring a competition over.
- `app/views/competitions/_admin_index_table.html.erb:56:                if competition.is_over?` - Admin stuff that I'm pretty sure the board doesn't care about when a competition is still ongoing.
- `app/views/competitions/_admin_index_table.html.erb:69:                if competition.is_over?` - Admin stuff that I'm pretty sure the board doesn't care about when a competition is still ongoing.
- `app/views/competitions/_index_competitions_list.html.erb:39:          is_over: c.is_over?,` - Just passing over the data for the javascript at the top of this list.
- `app/views/competitions/_index_table.html.erb:7:    <li class="list-group-item <%= competition.is_over? ? "past" : "not-past" %>">` - Some css classes for styling. See https://github.com/jfly/worldcubeassociation.org/blob/5a6e116fcae817be31477d1be7a6e04468a4ef9a/WcaOnRails/app/assets/stylesheets/competitions.scss#L103-L115.
- `app/views/competitions/_index_table.html.erb:9:        <% if competition.is_over? %>` - This controls what icons we show on https://www.worldcubeassociation.org/competitions. This is nice, but not really critical to get right.
- `app/views/competitions/_my_competitions_table.html.erb:58:              <% if competition.is_over? && competition.delegates.include?(current_user) && !competition.delegate_report.posted? %>` - This guards showing a warning icons to delegates about posting their delegate report.
- `app/views/delegates/stats.html.erb:21:        <% competitions = delegate.delegated_competitions.order(:year, :month, :day).select(&:is_over?) %>` - Controls what competitions are used for the display of https://www.worldcubeassociation.org/admin/delegates/. Also not really important which side we err on here.
- `spec/models/competition_spec.rb:46:    expect(competition.is_over?).to be false` - It's a spec, ignorable
- `spec/models/competition_spec.rb:216:      expect(competition.is_over?).to be true` - It's a spec, ignorable

From reading through our codebase, there are only 2 user actions that are controlled by whether or not a competition is over:

1. Being allowed to post a competition delegate report.
2. The tautological sounding competition validation `delegates_must_be_delegates` is only enforced for competitions that are not `is_over?`.

IMO, 2) doesn't really matter, and 1) is much better for our delegates if we aggressively delcare competitions as over, so I think it's okay to just redefine `is_over?` to be more aggressive (as I've done in this PR).

Alternatively you could argue that we shouldn't have a method called `is_over?`. Why are we defining methods we can't actually implement?* Instead, we could define `is_over_everywhere_on_earth` and `is_over_somewhere_on_earth` methods and thereby force the caller to think through which use case is most appropriate for them. I'd argue for this if I believed this distinction actually mattered, and if we didn't have a plan to eventually be able to implement an accurate `is_over?` method (someday we'll know competition schedules and timezones.

* This reminds me of one of my favorite methods I saw at my previous job:
```
def waitForKernelRoutes():
  sleep 10
```